lib/tesla/middleware/api_key_auth.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Tesla.Middleware.APIKeyAuth do
+  @moduledoc """
+  API key authentication middleware.
+
+  Adds a `{name, value}` tuple to headers or query params, based on `add_to` option.
+
+  ## Examples
+  ```
+  defmodule MyClient do
+    use Tesla
+    
+    # static configuration
+    plug Tesla.Middleware.APIKeyAuth, add_to: :header, name: "X-API-KEY", value: "api_key"
```

Comment:
I'm not sure there is any value in 

```ex
plug Tesla.Middleware.APIKeyAuth, add_to: :header, name: "X-API-KEY", value: "api_key"
```

compared to 

```ex
plug Tesla.Middleware.Headers, [{"X-API-KEY", "api_key"}]
```
or 

```ex
plug Tesla.Middleware.Query, [api_key: "api_key"]
```

Reply:
Looking at this couple of months later, I agree. Probably I was trying to fit it together with basic auth and bearer token, but it's really isn't adding anything besides an slightly more clear intent of what is the purpose of this header/param.